js/app/src/Blocks.svelte

```diff
@@ -258,13 +259,30 @@
 			if (api_recorder_visible) {
 				api_calls = [...api_calls, payload];
 			}
-			const submission = app
-				.submit(
+
+			let submission: SubmitReturn;
+			try {
+				submission = app.submit(
 					payload.fn_index,
 					payload.data as unknown[],
 					payload.event_data,
 					payload.trigger_id
-				)
+				);
+			} catch (e) {
+				const fn_index = 0; // Mock value for fn_index
```

Comment:
I created this PR as a proposal, but not sure if these implementation is correct. Plz modify it if there is a cleaner/more standard way.